cfme/configure/configuration/diagnostics_settings.py

```diff
@@ -358,6 +376,32 @@ def in_server_collect_logs(self):
         )
 
 
+class ServerCollectLogsSlaveView(ServerCollectLogsView):
+    @property
+    def is_displayed(self):
+        # select the first slave server (there's only one slave server in the tests,
+        # but appliance.server.slave_servers returns a list)
+        slave_server = self.context['object'].appliance.server.slave_servers[0]
+        return (
+            self.in_server_collect_logs and
+            self.title.text == 'Diagnostics Server "{name} [{sid}]"'.format(
+                name=slave_server.name,
+                sid=slave_server.sid)
+        )
+
+
+class ServerCollectLogsMasterView(ServerCollectLogsView):
+    @property
+    def is_displayed(self):
+        master_server = self.context['object'].appliance.server
+        return (
+            self.in_server_collect_logs and
+            self.title.text == 'Diagnostics Server "{name} [{sid}]" (current)'.format(
```

Comment:
Q: I have a distinct feeling of deja vu :).  Could this code be wrapped up into function or variable and used in all these views ?